nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    "#{self.class} has not implemented method '#{__method__}'"
```

Comment:
- Optional
Kindly use the `NotImplementedError` exception to handle the exception if the method has not been implemented. It is the more conventional way of indicating that a method has not been implemented. It raises an error when the method is called, which can be useful for debugging or when you want to ensure that a method is implemented in a subclass.